### DISCUSSION TITLE
[Projects Beta] Proposal: Colour scale fields

### DISCUSSION THREAD

**Discussion by brettwillis** [community/community#16940] [Projects and Issues]
There are several scale-type fields that are useful for project planning, for example:
- Effort (story points, time estimate)
- Priority

Currently I use a single-select type field with differenct options. So a "Priority" field might have `1 - Critical`, `2 - High`, `3 - Normal`, and `4 - Low` for example.

It would be easier to read such fields if I could also assign a colour scale, where `1 - Critical` would be red, and `4 - Low` would be blue. Currently you can partially simulate this by using labels (which are coloured) but they don't work with sorting, and they are all grouped into the labels field, not serparate columns for effort, priority, etc.

So my proposal is a new type of field called "Scale" or something with the following attributes:
- Inherently sortable by the order that they are defined in (so I can just say `Critical`, `High`..., etc instead of prefixing with numbers to enable sorting)
- Each value is coloured (like labels are), and this could be done by assigning a "start" and "end" colour, or by assigning individual colours (which is probaly more flexible and extensible to other fields and scenarios...)

Optionally, taking the proposal one step further, if we take "time estimate" as an example: it would be useful to have a numeric field with the following enhancements:
- Unit suffix, i.e. `2 hrs`
- Automatic colour scaling based on the minimum and maximum of all values in that column (similar to what you expect from colour scale conditional formatting in Excel)

**Comment by Jay-o-Way** [comment]
Interesting idea.

You might want to use emoji to signify meaning. It's more and more common. Also, numbers that are used for _priority_ are ambiguous. Does "zero" mean "life-or-death" priority or "no" priority..? I would advice to drop the numbers. For example:

**PRIORITY**

- HotFix  🔥
- High ❗
- Medium
- Low ❄
